src/fontra/views/editor/visualization-layer-definitions.js

```diff
@@ -851,6 +981,117 @@ registerVisualizationLayerDefinition({
+  selectionMode: "editing",
+  zIndex: 500,
+  screenParameters: {
+    smoothSize: 8,
+    strokeWidth: 1,
+    hoverStrokeOffset: 4,
+    underlayOffset: 2,
+    iconSize: 12,
+  },
+  colors: {
+    hoveredColorIcon: "#555",
+    hoveredColor: "#BBB",
+    selectedColor: "#000",
+    underColor: "#FFFA",
+  },
+  colorsDarkMode: {
+    hoveredColorIcon: "#BBB",
+    hoveredColor: "#BBB",
+    selectedColor: "#FFF",
+    underColor: "#0008",
+  },
+  draw: (context, positionedGlyph, parameters, model, controller) => {
+    const glyph = positionedGlyph.glyph;
+    const smoothSize = parameters.smoothSize;
+
+    const {
+      guideline: hoveredGuidelineIndices,
+      fontGuideline: hoveredFontGuidelineIndices,
+    } = parseSelection(model.hoverSelection);
+    const {
+      guideline: selectedGuidelineIndices,
+      fontGuideline: selectedFontGuidelineIndices,
+    } = parseSelection(model.selection);
+
+    // TODO: Font Guidelines
+
+    // Under layer
+    context.fillStyle = parameters.underColor;
+    for (const i of selectedGuidelineIndices || []) {
+      const guideline = glyph.guidelines[i];
+      if (!guideline) {
+        continue;
+      }
+      if (guideline.locked) {
+        _drawLockIcon(
+          context,
+          parameters,
+          guideline.x - parameters.iconSize / 2,
+          guideline.y + parameters.iconSize / 2
+        );
+      } else {
+        fillRoundNode(context, guideline, smoothSize + parameters.underlayOffset);
+      }
+    }
+
+    // Selected guideline
+    context.fillStyle = parameters.selectedColor;
+    for (const i of selectedGuidelineIndices || []) {
+      const guideline = glyph.guidelines[i];
+      if (!guideline) {
+        continue;
+      }
+      if (guideline.locked) {
+        context.strokeStyle = parameters.selectedColor;
+        _drawLockIcon(
+          context,
+          parameters,
+          guideline.x - parameters.iconSize / 2,
+          guideline.y + parameters.iconSize / 2
+        );
+      } else {
+        fillRoundNode(context, guideline, smoothSize);
+      }
+    }
+
+    // // Hovered guideline
+    context.strokeStyle = parameters.hoveredColor;
+    context.lineWidth = parameters.strokeWidth;
+    for (const i of hoveredGuidelineIndices || []) {
+      const guideline = glyph.guidelines[i];
+      if (!guideline) {
+        continue;
+      }
+      if (guideline.locked) {
+        context.strokeStyle = parameters.hoveredColorIcon;
+        _drawLockIcon(
+          context,
+          parameters,
+          guideline.x - parameters.iconSize / 2,
+          guideline.y + parameters.iconSize / 2
+        );
+      } else {
+        strokeRoundNode(context, guideline, smoothSize + parameters.hoverStrokeOffset);
+      }
+    }
+  },
+});
+
+function _drawLockIcon(context, parameters, x, y) {
+  context.save();
+  context.translate(x, y);
+  context.scale(parameters.iconSize / 24, (-1 * parameters.iconSize) / 24);
+  context.lineWidth = 2;
```

Comment:
I'm not sure of the status of the hover display behavior we discussed. I don't currently see it happening. Is it not yet implemented or is this lineWidth = 2 somehow mistaken?

Reply:
Do you see this:

https://github.com/googlefonts/fontra/assets/21055547/08a1afd6-3da6-4f16-a0bb-6afb1d25091c

Reply:
Now I do! I didn't last night.

However, when the guide is selected, and I hover, I do see the outline but the lock itself is displayed as if it's not selected. So I see no distinction between "hovered and selected" and "hovered and not selected".

Reply:
thanks. done.